### DISCUSSION TITLE
HA Docker Support for Rasperry Pi5 16kb page Size supported at any point ?

### DISCUSSION THREAD

**Discussion by Malte5768** [home-assistant/operating-system#3132] [General]
Hello there,
i have recently setup my rasperry pi 5 with HA as a docker container in order to learn docker and play with a thing or two. 
Im not using HA OS because im using my rasperry for other programming projects at the same time. 

For now i have editet the  /boot/firmware/configtxt  with kernel = kernel8.img because of the page size issue with different applications see https://github.com/raspberrypi/bookworm-feedback/issues/107#issuecomment-1773810662.

Is there anything planned in regards on fixing these issues in the future ?
Or have i simply overlooked a fix that already exists other than restricting the pi to 4kb page size? ^^

-im just curious here as im happily restricting my pi 5 to the 4kb page Size.

Looking forward to any comments and or answers to this.

-Malte

**Comment by agners** [comment]
I am aware that you don't want to use HAOS, but just for everyone else landing here, and since this is the HAOS discussion forum: Raspberry Pi 5 is supported by HAOS, you can [download images in the release section](https://github.com/home-assistant/operating-system/releases) and they will be soon available on the Raspberry Pi Imager as well.

You can also consider using OS virtualization via KVM/libvirt using the generic-aarch64 images :penguin: :muscle: 

That out of the way, the problem you describing was caused by the memory allocator used and will be solved with 2024.2, see also 
https://github.com/home-assistant/core/issues/105768#issuecomment-1900489496.

**Reply by Malte5768** [reply]
Woops wasnt ware of the extra forum for the docker version. 
Thanks for the info, i will most definitly look into it.
Have a good one !
